bin/amazon_ssa_agent

```diff
@@ -4,63 +4,8 @@ Dir.chdir(root_dir) { require 'bundler/setup' }
 $LOAD_PATH << File.join(root_dir, "lib")
 
 require 'optparse'
-require 'log4r'
-require 'log4r/configurator'
 require 'log_decorator'
-
 require 'aws-sdk'
-
-#
-# We must do this before anything accesses log4r.
-#
-module Log4r
-  Configurator.custom_levels(:DEBUG, :INFO, :WARN, :ERROR, :FATAL, :COPY)
```

Comment:
I don't recall why this was done, but do we no longer need the custom log level?

Reply:
No, the COPY level is specific to Log4r when it rolls over.  This instead is handled internally with Logger.

Reply:
Correction...it was a loglevel we used specifically to log when we roll the log, but it's not necessary.

Reply:
I thought it was used when we copied the log lines from the logs in S3 into our log files in the appliance. If my memory is correct, it enabled us to re-log the messages on the appliance without having duplicate information (time stamp, etc).

Reply:
I don't understand what you mean.  I don't see any code here, in https://github.com/ManageIQ/manageiq/pull/16058, nor in https://github.com/ManageIQ/manageiq-providers-amazon/pull/318 that does that.

Even so, the way it works is that as the logger reaches the maxsize, it will trip Logger's internal roller, and we just attach to that and write the file to S3.  Then when the file is rolled, you have a brand new, empty, file.

The only thing I could see where there might be a problem (if what you describe is eventually written) is if the agent process restarts itself within the container, then on shutdown it will write the file to S3, but it will still be present on the filesystem.  Then it will start again, and pick up where it left off in the log, meaning you'll get duplicates.  That should be easy to fix by just killing the log file after it writes to S3 in the at_exit, and I'll make a commit to do that.

Reply:
Added a commit to address that concern.

Reply:
As I recall, there was code in the original prototype that retrieved the logs from S3 (per scan) and wrote them the the evm.log via `log.copy ...`

Given we haven't really defined how we're going to handle the logs, we probably need not worry about it now.